dds_gen/src/lib.rs

```diff
@@ -15,7 +14,8 @@ pub fn compile_idl(idl_filepath: &Path) -> Result<String, String> {
         .next()
         .expect("Must contain a specification");
 
-    let mut output = String::new();
-    rust::generate_rust_source(parsed_idl, &mut output);
-    Ok(output)
+    let mut ctx = Context::<String>::default();
+    rust::generate_rust_source(parsed_idl, &mut ctx).map_err(|err| err.to_string())?;
```

Comment:
Once the other comments are processed I would say this should look like RustGenerator.generate_rust_source(parsed_idl)